apps/src/templates/census2017/NameInput.jsx

```diff
@@ -0,0 +1,37 @@
+import React, {Component, PropTypes} from 'react';
+import i18n from "@cdo/locale";
+import {styles} from './CensusForm';
+
+class NameInput extends Component {
+
+  static propTypes = {
+    setName: PropTypes.func,
+    name: PropTypes.string
+  }
+
+  sendToForm(event) {
+    this.props.setName(event.target.value);
+  }
+
+  render() {
+    return (
+     <div>
+       <label>
+         <div style={styles.question}>
+           {i18n.yourName()}
+         </div>
+         <input
+           type="text"
+           name="name_s"
+           value={this.props.name}
+           onChange={this.sendToForm.bind(this)}
```

Comment:
once you've bound `this` at the time you define `sendToForm`, it's best to pass this just as onChange={this.sendToForm}. This is so that we don't create extra objects (taking more memory, I think) every time we render.